Apcon modules

davidlee-ap opened the pull request:
##### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->
Add new modules to support APCON network devices

##### ISSUE TYPE
<!--- Pick one below and delete the rest -->
- New Module Pull Request

##### COMPONENT NAME
<!--- Write the short name of the module, plugin, task or feature below -->
lib/modules/network/apconos
lib/module_utils/network/apconos
lib/plugins/action/apconos_config.py
lib/plugins/action/apconos.py
lib/plugins/cliconf/apconos.py
lib/plugins/terminal/apconos.py

##### ADDITIONAL INFORMATION
<!--- Include additional information to help people understand the change here -->
<!--- A step-by-step reproduction of the problem is helpful if there is no related issue -->

<!--- Paste verbatim command output below, e.g. before and after your change -->
```paste below

```

Commit:
Add APCON modules

Commit:
Add APCON modules

Commit:
Fix cli regex

Commit:
Fix cli regex

Commit:
Pass Sanity Test

Commit:
Pass Sanity Test

mattclay commented on test/sanity/ansible-doc/skip.txt:
```diff
@@ -40,3 +40,4 @@ panos_security_policy
 s3
 vsphere_guest
 win_msi
+apconos_config
```
New modules should not be added to the skip list.

davidlee-ap replied:
I removed it.
Thanks!

davidlee-ap replied:
> 
> 
> New modules should not be added to the skip list.

I removed it.
Thanks!

Commit:
revise for passing sanity tests

kbreit commented on lib/ansible/modules/network/apconos/apconos_cert.py:
```diff
@@ -0,0 +1,190 @@
+DOCUMENTATION = """
+---
+module: apconos_cert
+version_added: "2.9.0"
```
It could also be 2.9 as no new modules are added on minor releases.

davidlee-ap replied:
I'll fix it!
Thanks!

kbreit commented on lib/ansible/modules/network/apconos/apconos_cert.py:
```diff
@@ -0,0 +1,190 @@
+author: "David Lee (@davidlee-ap)"
+short_description: install ssl ipv4 certificate on apcon network devices
+description:
+  - import and install ssl ipv4 certificate with specifying remote filename
```
Capitalize properly

davidlee-ap replied:
I'll fix it.
Thanks!

kbreit commented on lib/ansible/modules/network/apconos/apconos_cert.py:
```diff
@@ -0,0 +1,190 @@
+module: apconos_cert
+version_added: "2.9.0"
+author: "David Lee (@davidlee-ap)"
+short_description: install ssl ipv4 certificate on apcon network devices
```
Capitalize properly

davidlee-ap replied:
I'll fix it. Thanks!

kbreit commented on lib/ansible/modules/network/apconos/apconos_cert.py:
```diff
@@ -0,0 +1,190 @@
+options:
+  command:
+    description:
+      - currently it is not being used in apconos_cert module.
```
Why is it here then?

davidlee-ap replied:
Sanity test couldn't pass without it. It might be implemented in the future.

mattclay replied:
If it's not used by the module, remove it from the module spec instead of adding it to the docs.

davidlee-ap replied:
I'll remove it.
Thanks!

kbreit commented on lib/ansible/modules/network/apconos/apconos_cert.py:
```diff
@@ -0,0 +1,190 @@
+        retries=dict(default=10, type='int'),
+        interval=dict(default=1, type='int'),
+
+        ipaddress=dict(type='list'),
```
Why would this be a list instead of a `str`? Also, other modules normally use camel case. I'll defer to a core developer to see what they think about this convention.

davidlee-ap replied:
str should be OK, but it is supposed to be a list for the application.

kbreit replied:
What do you mean? How is this interfacing and what’s the application?

mattclay replied:
The module is only using `ipaddress[0]`, so this shouldn't be a list.

davidlee-ap replied:
> 
> 
> The module is only using `ipaddress[0]`, so this shouldn't be a list.

I'll fix it.
Thanks!

kbreit commented on lib/ansible/modules/network/apconos/apconos_cert.py:
```diff
@@ -0,0 +1,190 @@
+
+    spec.update(apconos_argument_spec)
+
+    module = AnsibleModule(argument_spec=spec, supports_check_mode=True)
```
Do you have anywhere in your code that's supporting check mode?

kbreit commented on lib/ansible/plugins/action/apconos_config.py:
```diff
@@ -0,0 +1,108 @@
+# Copyright (C) 2017 APCON.
```
Copyright is old.

davidlee-ap replied:
I'll fix it. Thanks!

danielmellado commented:
It'd be also great to have integration and unit tests. You can check for example /ansible/ansible/tree/devel/test/units/modules/network

Commit:
Add unit test for apcon_command module
Remove apconos_config.py apconos_update.py apconos_cert.py for now

Commit:
Add unit test for apcon_command module
Remove apconos_config.py apconos_update.py apconos_cert.py for now

davidlee-ap committed:
Merge branch 'devel' into apcon_modules

das7pad committed:
[s3_bucket] access the bucket encryption response gracefully (#59826)

Signed-off-by: Jakob Ackermann <das7pad@outlook.com>

samdoran committed:
Disable mongodb_shard and mongodb_replicaset tests

They are causing failures when run as part of the  entire group shippable/posix/group1/

jborean93 committed:
Fix ansible-galaxy import command to use argspec instead (#59898)

Akasurde committed:
Minor fixes in galaxy command for collection (#59846)

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

Commit:
Fix for santiy test

Commit:
Fix for santiy test

Commit:
Merge branch 'apcon_modules' of https://github.com/davidlee-ap/ansible into apcon_modules

Commit:
santiy

Commit:
rebase

Commit:
conflict resolved
